lib/nerves_hub_link/client.ex

```diff
@@ -76,6 +76,14 @@ defmodule NervesHubLink.Client do
   """
   @callback handle_error(any()) :: :ok
 
+  @doc """
+  Optional callback that may be supplied to allow
+  the `reboot` call to be replaced
+  """
+  @callback reboot() :: no_return()
+
+  @optional_callbacks [reboot: 0]
```

Comment:
```suggestion
  @callback handle_reboot() :: no_return()

  @optional_callbacks [handle_reboot: 0]
```

I would prob say this should be handled a bit like the other callbacks.

Also, do we catch any error and potentially force the reboot anyway?

Reply:
i don't really think that's necessary personally. I'm thinking that if a user implements their own `reboot` function, and they mess it up, it's either
1) in development, reboot would not be wanted if i were this developer
2) bad firmware that should crash and cause a quarantine anyway

Reply:
differ to @fhunleth what do you think?

Reply:
There's no error to catch since this has to happen in a separate process. 

Regarding misuse, the `fwup` configs can be bulletproofed to protect against bricking devices via double-upgrading. Some of them do, but it's one of those things that you have to think about and `fwup` is dumb by design on that aspect. I think this is ok.

I don't have opinions on handling the callback implementation. I do like consistency.

Reply:
I have one other update that i'd like to get in, so i'll add it for consistency.